api/v1alpha1/serviceintentions_types.go

```diff
@@ -148,13 +178,59 @@ func (in *SourceIntention) toConsul() *capi.SourceIntention {
 		Name:        in.Name,
 		Namespace:   in.Namespace,
 		Action:      in.Action.toConsul(),
+		Permissions: in.Permissions.toConsul(),
 		Description: in.Description,
 	}
 }
 
 func (in IntentionAction) toConsul() capi.IntentionAction {
 	return capi.IntentionAction(in)
 }
+
+func (in IntentionPermissions) toConsul() []*capi.IntentionPermission {
+	if in == nil {
+		return nil
+	}
```

Comment:
we probably don't need this since a nil array is an empty array.

Reply:
good catch!!